scripts/db/add-historical-chart-summaries.js

```diff
@@ -0,0 +1,211 @@
+        )
+        FOR webId IN webIds
+          FOR edge IN webToWebScans
+            FILTER edge._from == webId._id
+            RETURN MERGE(DOCUMENT(edge._to), {"domain": webId.domain})
+        `
+    )
+    .toArray()
+    .filter((scan) => scan.status !== "pending");
+
+  const uniqueWebDomains = removeDuplicates(dailyWebScans, "domain");
+
+  uniqueWebDomains.forEach((scan) => {
+    try {
+      const { httpsStatus = "info", hstsStatus = "info" } = scan.results.connectionResults;
+      const {
+        curveStatus = "info",
+        cipherStatus = "info",
+        protocolStatus = "info",
+        certificateStatus = "info",
+        sslStatus = "info",
+      } = scan.results.tlsResult;
+      if (scanList[scan.domain])
+        scanList[scan.domain] = {
+          ...scanList[scan.domain],
+          https: httpsStatus || "info",
+          hsts: hstsStatus || "info",
+          curve: curveStatus || "info",
+          cipher: cipherStatus || "info",
+          protocol: protocolStatus || "info",
+          certificate: certificateStatus || "info",
+          ssl: sslStatus || "info",
+        };
+    } catch (err) {
+      console.error(err);
+    }
+  });
+
+  const chartSummaries = {};
+  // generate chart summaries
+  for (const [key, val] of Object.entries(charts)) {
+    chartSummaries[key] = {
+      scanTypes: val,
+      pass: 0,
+      fail: 0,
+      total: 0,
+    };
+  }
+
+  let not_implemented_count = 0;
+  let assess_count = 0;
+  let deploy_count = 0;
+  let enforce_count = 0;
+  let maintain_count = 0;
+
+  // add chart summaries to the database
+  Object.keys(scanList).forEach((domain) => {
+    const statuses = scanList[domain];
+
+    for (const [_key, val] of Object.entries(chartSummaries)) {
+      let chart = val;
+      categoryStatus = [];
+      for (const scanType of chart.scanTypes) {
+        categoryStatus.push(statuses[scanType]);
+      }
+      if (categoryStatus.includes("fail")) {
+        chart.fail += 1;
+        chart.total += 1;
+      } else if (!categoryStatus.includes("info")) {
+        chart.pass += 1;
+        chart.total += 1;
+      }
+    }
+    switch (statuses.dmarcPhase) {
+      case "not implemented":
+        not_implemented_count += 1;
+        break;
+      case "assess":
+        assess_count += 1;
+        break;
+      case "deploy":
+        deploy_count += 1;
+        break;
+      case "enforce":
+        enforce_count += 1;
+        break;
+      case "maintain":
+        maintain_count += 1;
+        break;
+      default:
+        break;
+    }
+  });
+
+  const dmarcPhaseSummary = {
+    not_implemented: not_implemented_count,
+    assess: assess_count,
+    deploy: deploy_count,
+    enforce: enforce_count,
+    maintain: maintain_count,
+    total: not_implemented_count + assess_count + deploy_count + enforce_count + maintain_count,
+  };
+
+  db.chartSummaries.save({ date: currentDay.toLocaleDateString(), dmarcPhase: dmarcPhaseSummary, ...chartSummaries });
```

Comment:
This seems to be adding the dates differently than the service:

![image](https://github.com/canada-ca/tracker/assets/47400288/af1d1fca-842b-4c3a-bf52-07fd2133a837)